### PR TITLE
Fw rules md format

### PR DESCRIPTION
**Pull request by adisos**

### CODE REVIEW

File: network-config-analyzer/SchemeRunner.py [hunk 1]
```diff
@@ -270,7 +270,7 @@ def _run_semantic_diff(self, configs_array, output_configuration):
                     query_output += full_result.output_result
                 res += full_result.numerical_result
                 if not full_result.bool_result:
-                    query_output += full_result.output_explanation
+                    query_output += full_result.output_explanation + '\n'
         # print()
         #query_output += '\n'
```
Comment (zivnevo) [inline network-config-analyzer/SchemeRunner.py:275]: Please remove commented-out code

File: README.md [hunk 2]
```diff
@@ -54,7 +58,7 @@ Running with no command-line options at all is like running `nca.py --sanity k8s
 - `--daemon`\
   Run NCA as a daemon. Send and receive data using a REST API.
 - `--output_format <format>`\
-  Output format specification (txt/yaml/csv).\
+  Output format specification (txt/yaml/csv/md).\
   *default:* txt\
   *shorthand:* `-o`
 - `--file_out <file name>`\
```
Comment (zivnevo) [inline README.md:64]: Can you please also document the `--pr_url` flag I just added?
Its argument should be a URL pointing to the relevant `comments` resource in the GitHub API.
e.g., `https://api.github.com/repos/shift-left-netconfig/online-boutique/issues/1/comments`

File: network-config-analyzer/MinimizeFWRules.py [hunk 3]
```diff
@@ -690,20 +691,29 @@ def _get_fw_rules_content_str(self, query_name, req_format, add_txt_header, add_
...
+                    header_lines = [FWRule.rule_csv_header] + header_lines
+                else:
+                    header_lines = [FWRule.rule_csv_header, ['---']*len(FWRule.rule_csv_header)] + header_lines
             rules_list = header_lines + rules_list
             for row in rules_list:
-                row_str = ''
+                row_str = '' if is_csv else '|'
                 for elem in row:
-                    row_str += '\"' + elem + '\"' + ','
+                    row_str += ('\"' + elem + '\"' + ',') if is_csv else (elem + '|')
```
Comment (zivnevo) [inline network-config-analyzer/MinimizeFWRules.py:713]:
```suggestion
                    row_str += f'\"{elem}\",' if is_csv else f'{elem}|'
```

File: network-config-analyzer/MinimizeFWRules.py [hunk 4]
```diff
@@ -690,20 +691,29 @@ def _get_fw_rules_content_str(self, query_name, req_format, add_txt_header, add_
...
                 res = f'final fw rules for query: {query_name}:\n' + res
             return res
+
         elif req_format == 'yaml':
             yaml_query_content = [{'query': query_name, 'rules': rules_list}]
             res = yaml.dump(yaml_query_content, None, default_flow_style=False, sort_keys=False)
             return res
-        elif req_format == 'csv':
+
+        elif req_format == 'csv' or req_format == 'md':
```
Comment (zivnevo) [inline network-config-analyzer/MinimizeFWRules.py:700]:
```suggestion
        elif req_format in ['csv', 'md']:
```

File: network-config-analyzer/FWRule.py [hunk 5]
```diff
@@ -523,7 +518,7 @@ def get_rule_in_req_format(self, req_format, is_k8s_config):
         """
         if req_format == 'yaml':
             return self.get_rule_yaml_obj(is_k8s_config)
-        elif req_format == 'csv':
+        elif req_format == 'csv' or req_format == 'md':
             return self.get_rule_csv_row(is_k8s_config)
         elif req_format == 'txt':
             return self.get_rule_str(is_k8s_config)
```
Comment (zivnevo) [inline network-config-analyzer/FWRule.py:524]:
```suggestion
        if req_format in ['csv', 'md']:
            return self.get_rule_csv_row(is_k8s_config)
        if req_format == 'txt':
            return self.get_rule_str(is_k8s_config)
            
```

**Review by zivnevo — CHANGES_REQUESTED** [review]
Thanks Adi. Looks good.
A few minor comments below.

File: network-config-analyzer/SchemeRunner.py (hunk 1 above)
Comment (adisos) [reply network-config-analyzer/SchemeRunner.py]: done

File: README.md (hunk 2 above)
Comment (adisos) [reply README.md:64]: done

File: network-config-analyzer/MinimizeFWRules.py (hunk 3 above)
Comment (adisos) [reply network-config-analyzer/MinimizeFWRules.py]: done

File: network-config-analyzer/MinimizeFWRules.py (hunk 4 above)
Comment (adisos) [reply network-config-analyzer/MinimizeFWRules.py]: done

File: network-config-analyzer/FWRule.py (hunk 5 above)
Comment (adisos) [reply network-config-analyzer/FWRule.py]: done

**Review by zivnevo — APPROVED** [review]
ok to merge

**Pull request merged**